[Refactoring] make transformations return whether graph is modified

penguinwu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#54777 [Refactoring] make transformations return whether graph is modified**

Summary: Updated RemoveListMutation, PeepholeOptimizedListIdoms,
LoopUnrolling, PeepholeOptimization to return whether graph is
modified after transformation, PeepholeAliasSensitivity

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27412105](https://our.internmc.facebook.com/intern/diff/D27412105)

Commit:
[Refactoring] make transformations return whether graph is modified

Summary: Updated RemoveListMutation, PeepholeOptimizedListIdoms,
LoopUnrolling, PeepholeOptimization to return whether graph is
modified after transformation, PeepholeAliasSensitivity

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

penguinwu commented:
This PR depends on #54640 which contains a fix in `jit_utils.py`. Will rebase once #54640 is merged in. Without it, there are 17 additional failures due to return type mismatch in `run_pass`

eellison commented on torch/csrc/jit/passes/loop_unrolling.cpp:
```diff
@@ -360,9 +365,12 @@ Node* PeelLoop(Node* n, size_t times) {
-  EliminateDeadCode(graph);
+bool UnrollLoops(std::shared_ptr<Graph>& graph) {
+  bool changed = UnrollLoops(graph->block());
+  if (changed) {
```
nice

eellison reviewed (approved):
Nice, looks great!! thanks for doing this!! It might be nice to add a comment to the header files that the passes return whether a change was made

Commit:
Update on "[Refactoring] make transformations return whether graph is modified"


Summary: Updated RemoveListMutation, PeepholeOptimizedListIdoms,
LoopUnrolling, PeepholeOptimization to return whether graph is
modified after transformation, PeepholeAliasSensitivity

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

Krovatkin reviewed (approved):
:shipit:

Commit:
Update on "[Refactoring] make transformations return whether graph is modified"


Summary: Updated RemoveListMutation, PeepholeOptimizedListIdoms,
LoopUnrolling, PeepholeOptimization to return whether graph is
modified after transformation, PeepholeAliasSensitivity

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

Commit:
Update on "[Refactoring] make transformations return whether graph is modified"


Summary: Updated RemoveListMutation, PeepholeOptimizedListIdoms,
LoopUnrolling, PeepholeOptimization to return whether graph is
modified after transformation, PeepholeAliasSensitivity

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27412105](https://our.internmc.facebook.com/intern/diff/D27412105)

[ghstack-poisoned]
